fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -320,6 +328,53 @@ public TListTableStatusResult listTableStatus(TGetTablesParams params) throws TE
         return result;
     }
 
+    @Override
+    public TListTaskInfoResult showTasks(TShowTasksParams params) throws TException {
+        LOG.debug("get show task request: {}", params);
+        TListTaskInfoResult result = new TListTaskInfoResult();
+        List<TTaskInfo> tasksResult = Lists.newArrayList();
+        result.setTables(tasksResult);
+
+        TaskManager taskManager = GlobalStateMgr.getCurrentState().getTaskManager();
+        List<Task> taskList = taskManager.showTask().stream()
+                .filter(u -> u.getDbName().equals(params.db)).collect(Collectors.toList());
```

Comment:
Filter db in `taskManager.showTask`？

Reply:
Yes, Task is of DB granularity, and TaskManager is not optimized for DB queries. BE is sent in the order of DB, so FE is filtered here.